### PR TITLE
Add notes about RAM limits

### PR DESCRIPTION
**Pull request by Hwurzburg**
Please add any other suggestions for temporary RAM savings and/or symptoms on the bench....I did not put a note about examining MEMINFO because there is no definite number to tell them if thats the problem, I believe, due to varying setups and whats not starting...

### CODE REVIEW

**Comment by peterbarker** [comment]
Flash and RAM are orthogonal.  I would guess that several full-features boards will also suffer from the lack-of-RAM problem.

I'd guess people have noticed it on these boards as they're more likely to have the fft and OSD stuff enabled....

One other option to avoid this is to calibrate the compasses sequentially rather than in parallel.  Obviously that's a bunch more tumbling the user would have to do.

Do we only get a one-short message if terrain allocation fails?  I thought we got a prearm for that?

**Comment by Hwurzburg** [comment]
the f4s have less ram I am told

my failure happened with only one compass, one gps...nothing else

I got NO message about terrain not downloading from the GCS, just remained pending after GPS lock, and MAVftp did not work....

got compass to cal by disabling logging temporarily....

ultimately, I found that I had IMU batch sampling active which consumes about 6K....had copied a previous TVBS with it running (Pete said he keeps it always enabled...but that is obviously not good  now)

**Comment by peterbarker** [comment]
On Sun, 27 Sep 2020, Henry Wurzburg wrote:

> the f4s have less ram I am told

Sure, but several F4s are full-features.

> I got NO message about terrain not downloading from the GCS, just remained pending after GPS lock, and MAVftp did not work....

I wonder if we should just go into the poorly-named "config error loop" if 
we get these allocation errors on startup.  We *could* add prearm checks, 
but that would be a lot more code.

> ultimately, I found that I had IMU batch sampling active which consumes about 6K....had copied a previous TVBS with it running (Pete said he keeps it
> always enabled...but that is obviously not good now)

Dumping the dynamic allocation tables from ChibiOS into a ftp'able file 
would be cool.  Of course, if FTP fails to allocate.... :-)

**Comment by Hwurzburg** [comment]
@peterbarker any problems merging this as is?

File: common/source/docs/common-limited-firmware.rst [hunk 1]
```diff
@@ -76,3 +76,19 @@ The missing features are listed below.
 
    -  Parachute
    -  Sprayer
+
+RAM Limitations
+===============
+
+Occasionally, in the above boards, RAM will become limited with enabled options in the firmware also. This is manifested by various symptoms:
```
Comment (kd0aij) [inline common/source/docs/common-limited-firmware.rst:83]:
```suggestion
There may be insufficient RAM available in some flight controllers to support all enabled firmware features. This is manifested by various symptoms:
```

File: common/source/docs/common-limited-firmware.rst [hunk 2]
```diff
@@ -76,3 +76,19 @@ The missing features are listed below.
...
+
+- MAVFtp does not work for fast connections to GCS
```
Comment (kd0aij) [inline common/source/docs/common-limited-firmware.rst:85]:
```suggestion
- MAVFtp does not work: parameter fetches to GCS will be slow
```

**Comment by kd0aij** [comment]
If MEMINFO shows less than 16K RAM free at bootup, there's not enough to run mavftp.
It seems that mavproxy doesn't automatically do a param fetch on connection, so that's one way to measure it. Otherwise, enabling log_disarmed should provide the necessary info. 
Ideally though, when something fails to start or run due to insufficient RAM it should emit a warning message.

File: common/source/docs/common-limited-firmware.rst (hunk 1 above)
Comment (peterbarker) [reply common/source/docs/common-limited-firmware.rst:83]: Should probably be simply, `Some symptoms are`.

(the symptoms would be of the manifestation, but that's implied).

File: common/source/docs/common-limited-firmware.rst (hunk 2 above)
Comment (peterbarker) [reply common/source/docs/common-limited-firmware.rst:85]: Should be `MAVFTP`  (FTP and MAV both being acronyms).

**Review by peterbarker — COMMENTED** [review]
SRTL on Copter (set points to zero)
Scripting (disable it)
Having both EK2 and EK3 (set (e.g.) EK2_ENABLE to zero)

**Comment by kd0aij** [comment]
Related issue and start of a PR: https://github.com/ArduPilot/ardupilot/pull/15468

File: common/source/docs/common-limited-firmware.rst (hunk 1 above)
Comment (Hwurzburg) [reply common/source/docs/common-limited-firmware.rst]: done

File: common/source/docs/common-limited-firmware.rst (hunk 2 above)
Comment (Hwurzburg) [reply common/source/docs/common-limited-firmware.rst]: done

**Comment by Hwurzburg** [comment]
@peterbarker did not mention scripting nor EKF...the 1Mb boards are most likely to have the issue now...2MB boards have more ram, in addition to the additional flash

**Comment by Hwurzburg** [comment]
getting more people wit this issue....going to merge this....any further issues, we can modify

**Pull request merged**